src/routers/documentationRouter.ts

```diff
@@ -1,13 +1,49 @@
 import express from 'express';
 import swaggerUi from 'swagger-ui-express';
 import documentation from '../openapi.json';
+import { issuer } from '../auth/jwtOptions';
 import { isJsonObject } from '../types';
+import { InternalValidationError } from '../errors';
+import type { SwaggerUiOptions } from 'swagger-ui-express';
+
+// Pre-fill expected OIDC/OAuth values in the documentation UI.
+const options: SwaggerUiOptions = {
+  // The `oauth` key in options is specific to `swagger-ui-express`.
+  // See https://github.com/scottie1984/swagger-ui-express/issues/5 and
+  // https://github.com/scottie1984/swagger-ui-express/blob/5cca5de75f7a42285acf76658734c1cf1ad01c61/index.js#L130
+  // The keys inside are from `swagger-ui` proper.
+  // See https://swagger.io/docs/open-source-tools/swagger-ui/usage/oauth2/.
+  swaggerOptions: {
+    oauth: {
+      clientId: 'pdc-openapi-docs',
+      realm: 'pdc',
+      scopes: [
+        'openid',
+        'roles',
+        'profile',
+      ],
+    },
+  },
+};
 
 const documentationRouter = express.Router();
 
 documentationRouter.use('/', swaggerUi.serve);
 if (isJsonObject(documentation)) {
-  documentationRouter.get('/', swaggerUi.setup(documentation));
+  // Substitute the authorization server URLs using the issuer loaded by `../auth/jwtOptions`.
+  const authCodeWithNewIssuer = {
+    ...documentation.components.securitySchemes.auth.flows.authorizationCode,
+    authorizationUrl: `${issuer}/protocol/openid-connect/auth`,
+    tokenUrl: `${issuer}/protocol/openid-connect/token`,
+    refreshUrl: `${issuer}/protocol/openid-connect/token`,
+  };
+  documentation.components.securitySchemes.auth.flows.authorizationCode = authCodeWithNewIssuer;
+  documentationRouter.get('/', swaggerUi.setup(documentation, options));
+} else {
+  throw new InternalValidationError(
```

Comment:
Now that I look more closely, I think I do actually agree with Codecov here.

Adding an error handler for improperly formatted JSON is a great idea, and I'm +1 to stopping the service over it, but it's not integral to setting up auth. Let's split it out to its own commit, put the `throw` logic into the `isJsonObject` function, rename it to something like `validateOpenApiConfiguration`, and add to the existing tests!

Then, later, let's add - and test - a function that will take that already-validated configuration and substitute in the issuer. Something like:

```ts
const getSwaggerDocsWithIssuer = (documentation: JsonObject, issuer: string): JsonObject => {
  const authCodeWithNewIssuer...
  return documentationWithNewIssuer;
};
documentationRouter.use('/', swaggerUi.serve);
documentationRouter.get('/', swaggerUi.setup(getSwaggerDocsWithIssuer(documentation, issuer), options));
```

That should be a bit easier to read, much easier to test, and will address the Codecov complaint this change introduced.

Reply:
Is it OK to have that commit in a separate PR? For now I can simply remove it.

Reply:
I removed the `else` in the most recent push.